### ISSUE TITLE
C++ support in Azure Functions?

### ISSUE THREAD

**Issue by christopheranderson** [Azure/azure-functions-host#1469]
_From @iszczesniak on March 24, 2017 9:10_

Hi,

I would like to use C++ REST SDK to develop my code in C++ and run it as an Azure Function.

Are there plans to support C++ for Azure Functions?

Best,
Irek

_Copied from original issue: Azure/Azure-Functions#224_

**Comment by PsychoXIVI** [comment]
How about using `DllImport` in C# for externing C++ code when necessary?

**Comment by esride-jts** [comment]
Come on show - some love to the native side . . .

**Comment by scalablecory** [comment]
Lack of C++ support is the only hurdle keeping my business in AWS Lambda.

**Comment by gisfromscratch** [comment]
Any progress on this? We are still waiting before migrating our C++ based location analytics toolset to azure.

**Comment by andrunl** [comment]
AWS supports this?
Sad, as our code is MS C++, would be weird to move it to AWS as Microsoft does not want to support its own codebase.

**Comment by sgreenlay** [comment]
You can now supply [your own container](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-function-linux-custom-image) to run in Azure Functions. I've used this to run unsupported languages like Go and Swift in Functions and you could use them to run C++.

**Comment by Xyncgas** [comment]
Webassembly.

**Comment by Xyncgas** [comment]
make it happen bill gates

**Comment by Xyncgas** [comment]
please

**Comment by balag0** [comment]
@anthonychu  @pragnagopa 
Can custom handlers help here https://docs.microsoft.com/en-us/azure/azure-functions/functions-custom-handlers

**Comment by anthonychu** [comment]
Yes. We don't have an C++ example but if you can wrap your code in an http server, you can run it as a custom handler.

For WebAssembly, the simplest way is probably to call it from a Node.js function. I also have an example as a custom handler here: https://github.com/anthonychu/azure-functions-wagi-sample

**Comment by specious** [comment]
> @anthonychu @pragnagopa Can custom handlers help here https://docs.microsoft.com/en-us/azure/azure-functions/functions-custom-handlers

What we need is the equivalent of [awslabs/aws-lambda-cpp](https://github.com/awslabs/aws-lambda-cpp) for Azure Functions.

Here's an [example](https://github.com/littermap/littermap-aws-backend/blob/1b16f497497bf44da07823314ee00da6207942a6/functions/scale-image/src/main.cpp#L73) of its use in an AWS-based project.

**Comment by gisfromscratch** [comment]
@anthonychu any progress at least a discussion? Currently, we ship our serverless functions using Python and compile the C++ core libraries to native Python modules using [pybind11](https://github.com/pybind/pybind11). Our team did some prototypes using Rust functions, as well. But, I don't think we are going to migrate our core libs using Rust in the next months.
